lib/ansible/modules/database/postgresql/postgresql_sequence.py

```diff
@@ -0,0 +1,646 @@
+
+        if self.module.params.get('start'):
+            query.append('START WITH %s' % self.module.params['start'])
+
+        if self.module.params.get('cache'):
+            query.append('CACHE %s' % self.module.params['cache'])
+
+        if self.module.params.get('cycle'):
+            query.append('CYCLE')
+
+        return exec_sql(self, ' '.join(query), ddl=True)
+
+    def drop(self):
+        """Implements DROP SEQUENCE command behavior."""
+        query = ['DROP SEQUENCE']
+        query.append(self.__add_schema())
+
+        if self.module.params.get('cascade'):
+            query.append('CASCADE')
+
+        return exec_sql(self, ' '.join(query), ddl=True)
+
+    def rename(self):
+        """Implements ALTER SEQUENCE RENAME TO command behavior."""
+        query = ['ALTER SEQUENCE']
+        query.append(self.__add_schema())
+        query.append('RENAME TO %s' % pg_quote_identifier(self.module.params['rename_to'], 'sequence'))
+
+        return exec_sql(self, ' '.join(query), ddl=True)
+
+    def set_owner(self):
+        """Implements ALTER SEQUENCE OWNER TO command behavior."""
+        query = ['ALTER SEQUENCE']
+        query.append(self.__add_schema())
+        query.append('OWNER TO %s' % pg_quote_identifier(self.module.params['owner'], 'role'))
+
+        return exec_sql(self, ' '.join(query), ddl=True)
+
+    def set_schema(self):
+        """Implements ALTER SEQUENCE SET SCHEMA command behavior."""
+        query = ['ALTER SEQUENCE']
+        query.append(self.__add_schema())
+        query.append('SET SCHEMA %s' % pg_quote_identifier(self.module.params['newschema'], 'schema'))
+
+        return exec_sql(self, ' '.join(query), ddl=True)
+
+    def __add_schema(self):
+        return '.'.join([pg_quote_identifier(self.schema, 'schema'),
+                        pg_quote_identifier(self.name, 'sequence')])
+
+
+# ===========================================
+# Module execution.
+#
+
+def main():
+    argument_spec = postgres_common_argument_spec()
+    argument_spec.update(
+        sequence=dict(type='str', required=True, aliases=['name']),
+        state=dict(type='str', default='present', choices=['absent', 'present']),
+        data_type=dict(type='str', choices=['bigint', 'integer', 'smallint']),
+        increment=dict(type='int'),
+        minvalue=dict(type='int', aliases=['min']),
+        maxvalue=dict(type='int', aliases=['max']),
+        start=dict(type='int'),
+        cache=dict(type='int'),
+        cycle=dict(type='bool'),
+        schema=dict(type='str', default='public'),
+        cascade=dict(type='bool'),
+        rename_to=dict(type='str'),
+        owner=dict(type='str'),
+        newschema=dict(type='str'),
+        db=dict(type='str', default='', aliases=['login_db', 'database']),
+        session_role=dict(type='str'),
+    )
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        mutually_exclusive=[
+            ['absent', 'data_type'],
```

Comment:
There is no option called `absent`?

Reply:
You are right. I thought I can use it for `state = absent`, but I think not. I have to fix it in another way

Reply:
I think I did not need the check at all. If `state = absent` it will ignore all other arguments.
So I just remove this here.

Reply:
Yep, simply ignoring all other options when `state == 'absent'` is the easiest solution (that's what most other modules do, too).